File: dotnet/coverage-tests/test/Playwright/PlaywrightDriverBuilder.cs
using System;
using System.Collections.Generic;
using System.Threading.Tasks;
using Microsoft.Playwright;

namespace Applitools.Generated.Playwright.Tests
{
    public class PlaywrightDriverBuilder
    {
        private bool headless_ = true;
        private bool legacy_ = false;
        private bool executionGrid_ = false;
        private string browser_ = "chrome";
        private IPlaywright playwright_;

        private static readonly Dictionary<string, IPlaywrightBuilder> playwrightBuilders_ = new()
        {
            { "chrome", new ChromeBuilder() },
            { "firefox", new FirefoxBuilder() }
        };

        public PlaywrightDriverBuilder Headless(bool headless)
        {
            headless_ = headless;
            return this;
        }

        public PlaywrightDriverBuilder Browser(string browser)
        {
            browser_ = browser;
            return this;
        }

        public PlaywrightDriverBuilder Legacy(bool legacy)
        {
            legacy_ = legacy;
            return this;
        }

        public PlaywrightDriverBuilder ExecutionGrid(bool executionGrid)
        {
            executionGrid_ = executionGrid;
            return this;
        }

        public IPage Build()
        {
            async Task<IPage> BuildAsync()
            {
                playwright_ = await Microsoft.Playwright.Playwright.CreateAsync();
                if (playwrightBuilders_.TryGetValue(browser_, out IPlaywrightBuilder builder))
                {
                    return builder.Build(playwright_, headless_, legacy_, executionGrid_);
                }

                throw new Exception($"builder for {browser_} not found");
            }

            return BuildAsync().GetAwaiter().GetResult();
        }

        public void Quit()
        {
            playwright_?.Dispose();
        }
    }
}

File: java/eyes.playwright.java/src/test/java/coverage/drivers/playwright/PlaywrightDriverBuilder.java
package coverage.drivers.playwright;

import com.microsoft.playwright.Page;
import com.microsoft.playwright.Playwright;


import java.net.MalformedURLException;
import java.util.HashMap;

public class PlaywrightDriverBuilder {

    protected boolean headless = true;
    protected boolean legacy = false;
    protected boolean executionGrid = false;
    protected String browser = "chrome";
    protected Playwright playwright;

    static HashMap<String, PlaywrightBuilder> playwrightBuilders = new HashMap<String, PlaywrightBuilder>() {{
        put("chrome", new ChromeBuilder());
        put("firefox", new FirefoxBuilder());
    }};

    public PlaywrightDriverBuilder headless(boolean headless) {
        this.headless = headless;
        return this;
    }

    public PlaywrightDriverBuilder browser(String browser) {
        this.browser = browser;
        return this;
    }

    public PlaywrightDriverBuilder legacy(boolean legacy) {
        this.legacy = legacy;
        return this;
    }

    public PlaywrightDriverBuilder executionGrid(boolean executionGrid) {
        this.executionGrid = executionGrid;
        return this;
    }

    public Page build() throws MalformedURLException {
        playwright = Playwright.create();
        PlaywrightBuilder builder = playwrightBuilders.get(browser);
        return builder.build(playwright, headless, legacy, executionGrid);
    }

    public void quit() {
        if (playwright != null) {
            playwright.close();
        }
    }
}
